fs/metrics/common/metrics.go

```diff
@@ -226,3 +232,36 @@ func AddBytesCount(operation string, layer digest.Digest, bytes int64) {
 func AddImageOperationCount(operation string, image digest.Digest, count int32) {
 	imageOperationCount.WithLabelValues(operation, image.String()).Add(float64(count))
 }
+
+// ListenForFuseFailure infinitely listens for any FUSE failure.
+// If one occurs, it increments the `FuseFailureState` metric and
+// sleeps for a time duration. This should be run at an FS level,
+// in a separate routine so that it does not block the main
+// thread.
+func ListenForFuseFailure(ctx context.Context) {
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-FuseFailureSignal:
+			IncOperationCount(FuseFailureState, digest.Digest(""))
+			select {
```

Comment:
Perhaps consider using an anonymous function for the nested select block.

e.g. 
```go
incrementWithBackoff := func(ctx context.Context) {
    IncOperationCount(FuseFailureState, digest.Digest(""))
    
    select {
    case <-ctx.Done():
        return
    case <-time.After(...):
        return
    }
}

for {
    select {
    case <-ctx.Done():
        return
    case <-FuseFailureSignal:
        incrementWithBackoff(...)
    }
}
```

Functionally the same, but gives a name to the inner select. (minor nit/not required)